sopel/config/types.py

```diff
@@ -5,24 +5,45 @@
 are available in this module) are assigned as attributes. These descriptors
 define how to read values from, and write values to, the config file.
 
-As an example, if one wanted to define the ``[spam]`` section as having an
-``eggs`` option, which contains a list of values, they could do this:
+As an example, if one wanted to create a section with an ``eggs`` option, which
+contains a list of values, they can create a subclass of
+:class:`StaticSection`::
 
-    >>> class SpamSection(StaticSection):
+    >>> from sopel.config import types
+    >>> class SpamSection(types.StaticSection):
     ...     eggs = ListAttribute('eggs')
     ...
-    >>> SpamSection(config, 'spam')
-    >>> print(config.spam.eggs)
+
+To use that section for the configuration file at ``/etc/sopel/default.cfg``,
+they must define the ``[spam]`` section on the settings with
+:meth:`~sopel.config.Config.define_section`::
+
+    >>> from sopel import config
+    >>> settings = config.Config('/etc/sopel/default.cfg')
+    >>> settings.define_section('spam', SpamSection)
+
+Without that, Sopel's configuration won't recognize the type of ``spam.eggs``
+properly, which can lead to invalid values.
+
+.. note::
+
+    The :meth:`~sopel.config.Config.define_section` is usually used in the
+    plugin's ``setup`` and ``configure`` functions. You can read more about
+    the ways to :ref:`configure your plugin <plugin-anatomy-config>` in the
+    plugin section of the documentation.
```

Comment:
Adding "method" is grammatically necessary, but modifying the hyperlink text was a style choice. :)

```suggestion
    The :meth:`~sopel.config.Config.define_section` method is usually used
    in the plugin's ``setup`` and ``configure`` functions. You can read more
    about the :ref:`ways to configure your plugin <plugin-anatomy-config>`
    in the plugin section of the documentation.
```

Note: Width of re-wrap is "close enough" but not verified in a real text editor.